astropy/io/fits/connect.py

```diff
@@ -302,3 +306,202 @@ def write_table_fits(input, output, overwrite=False):
 io_registry.register_reader('fits', Table, read_table_fits)
 io_registry.register_writer('fits', Table, write_table_fits)
 io_registry.register_identifier('fits', Table, is_fits)
+
+
+def read_data_fits(filename, ext_data=0, ext_meta=0, ext_mask='mask',
+                   ext_uncert='uncert', kw_unit='bunit', copy=False,
+                   dtype=None, **kwargs_for_open):
+    """
+    Read data from a FITS file and wrap the contents in a \
+    `~astropy.nddata.NDData`.
+
+    Parameters
+    ----------
+    filename : str and other types
+        see :func:`~astropy.io.fits.open` what possible types are allowed.
+
+    ext_data, ext_meta, ext_mask, ext_uncert : str or int, optional
+        Extensions from which to read ``data``, ``meta``, ``mask`` and
+        ``uncertainty``.
+        Default is ``0`` (data), ``0`` (meta), ``'mask'`` (mask) and
+        ``'uncert'`` (uncertainty).
+
+    kw_unit : str or None, optional
```

Comment:
I had one where there was no `BUNIT` but a `DATAUNIT`. I don't remember which one this was. Unfortunatly I deleted the test that checked my fits files with the reader. I don't know if the `DATAUNIT` was meant as unit for the data neither if it was added manually or by some 3rd party software or at the archive.

I don't know if it's common though. But the default is `BUNIT` and I've included in the documentation a big red box that states one shouldn't mess with the keywords, if not necessary: http://astropy-mseifer.readthedocs.io/en/tmp_nddata_final/nddata/mixins/ndio.html